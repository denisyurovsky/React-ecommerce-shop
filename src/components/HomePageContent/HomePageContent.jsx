import React, { useState } from 'react';

import { CardArea } from './CardArea/CardArea';
import { CardsContainer } from './CardsContainer/CardsContainer';
import { SearchArea } from './SearchArea/SearchArea';

import styles from './HomePageContent.module.scss';

export const initialCards = [
  {
    image: 'url',
    name: 'name11',
    type: 'First category',
    creationDate: '000',
    updateDate: '111',
    price: '300$',
    id: 1,
  },
  {
    image: 'url',
    name: 'name1',
    type: 'First category',
    creationDate: '000',
    updateDate: '111',
    price: '300$',
    id: 2,
  },
  {
    image: 'url',
    name: 'name2',
    type: 'Second category',
    creationDate: '000',
    updateDate: '111',
    price: '300$',
    id: 3,
  },
  {
    image: 'url',
    name: 'name3',
    type: 'Third category',
    creationDate: '000',
    updateDate: '111',
    price: '300$',
    id: 4,
  },
  {
    image: 'url',
    name: 'ebyb',
    type: 'Third category',
    creationDate: '000',
    updateDate: '111',
    price: '300$',
    id: 5,
  },
];

export const HomePageContent = () => {
  const [cards, setCards] = useState(initialCards);

  return (
    <section className={styles.container}>
      <h2 data-testid="title" className={styles.header}>
        Welcome to Born2Die Market!
      </h2>
      <SearchArea setCards={setCards} initialCards={initialCards}></SearchArea>
      <CardArea>
        <CardsContainer cards={cards}></CardsContainer>
      </CardArea>
    </section>
  );
};
