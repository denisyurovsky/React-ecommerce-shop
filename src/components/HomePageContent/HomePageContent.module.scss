@import "../../assets/styles/colors.scss";

.container {
  width: calc(100% - 200px);
  position: relative;
  padding-top: 40px;
}

.header {
  color: $primaryMain;
  font-size: 40px;
  font-weight: 400;
}

@media all and (max-width: 700px) {
  .header {
    font-size: 30px;
  }
}

@media all and (max-width: 700px) {
  .container {
    width: calc(100% - 100px);
  }
}

@media all and (max-width: 500px) {
  .header {
    font-size: 25px;
  }

  .container {
    width: calc(100% - 10px);
  }
}
