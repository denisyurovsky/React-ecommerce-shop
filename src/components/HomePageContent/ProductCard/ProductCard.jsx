import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import { indigo } from '@mui/material/colors';
import Typography from '@mui/material/Typography';
import PropTypes from 'prop-types';
import React from 'react';

import styles from './ProductCard.module.scss';

export const ProductCard = ({ card }) => {
  const { image, name, creationDate, updateDate, price, type } = card;

  return (
    <Card className={styles.card}>
      <CardMedia
        component="img"
        image={image}
        alt={name}
        className={styles.cardImage}
      />
      <CardContent>
        <Typography color={indigo[500]} gutterBottom variant="h5">
          {name}
        </Typography>
        <Typography variant="overline">{type}</Typography>
        <Typography variant="body2" color="text.secondary">
          Created:{creationDate}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          Updated: {updateDate}
        </Typography>
      </CardContent>
      <CardActions className={styles.cardActions}>
        <Typography variant="h5" component="p">
          {price}
        </Typography>
        <Button variant="contained" className={styles.button}>
          + add to cart
        </Button>
      </CardActions>
    </Card>
  );
};

ProductCard.propTypes = {
  card: PropTypes.shape({
    image: PropTypes.string,
    name: PropTypes.string,
    creationDate: PropTypes.string,
    updateDate: PropTypes.string,
    price: PropTypes.string,
    type: PropTypes.string,
    id: PropTypes.number,
  }).isRequired,
};
