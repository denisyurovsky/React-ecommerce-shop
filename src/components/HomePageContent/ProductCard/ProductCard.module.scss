@import "../../../assets/styles/colors.scss";

.cardName {
  color: $primaryMain;
}

.card {
  width: 30%;
  padding: 10px 10px 20px;
  margin-right: 1%;
  margin-top: 20px;
  box-sizing: border-box;
}

@media all and (max-width: 950px) {
  .card {
    width: 45%;
  }
}

@media all and (max-width: 600px) {
  .card {
    width: 80%;
  }
}

.cardImage {
  width: 85% !important;
  height: 200px;
  margin-left: auto;
  margin-right: auto;
}

.cardActions {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.button {
  text-transform: capitalize;
}
