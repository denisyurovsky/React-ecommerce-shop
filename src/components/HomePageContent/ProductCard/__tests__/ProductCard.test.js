import { render } from '@testing-library/react';
import React from 'react';

import { initialCards } from '../../HomePageContent.jsx';
import { ProductCard } from '../ProductCard';

describe('ProductCard component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = render(<ProductCard card={initialCards[0]} />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
