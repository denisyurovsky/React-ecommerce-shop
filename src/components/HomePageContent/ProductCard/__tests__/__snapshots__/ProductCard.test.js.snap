// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="name11"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img cardImage css-o69gx8-MuiCardMedia-root"
      src="url"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-15k7a75-MuiTypography-root"
      >
        name11
      </h5>
      <span
        class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
      >
        First category
      </span>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Created:000
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Updated: 111
      </p>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing cardActions css-1t6e9jv-MuiCardActions-root"
    >
      <p
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        300$
      </p>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        + add to cart
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
